### PR TITLE
feat: add file ram bundle support

### PR DESCRIPTION
**Pull request by zamotany**
https://github.com/callstack/haul/projects/2#card-21105966

### CODE REVIEW

File: packages/haul-ram-bundle-webpack-plugin/src/FileRamBundle.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+  constructor(
+    bootstrap: string,
+    modules: Module[],
+    sourceMap: boolean = false
+  ) {
+    this.bootstrap = bootstrap;
+    this.modules = modules;
+    this.sourceMap = sourceMap;
+  }
```
Comment (dratwas) [inline packages/haul-ram-bundle-webpack-plugin/src/FileRamBundle.ts]: WDYT about ?
```js
constructor(
    public bootstrap: string,
    public modules: Module[],
    public sourceMap: boolean = false
  )
```

File: packages/haul-ram-bundle-webpack-plugin/src/__tests__/FileRamBundle.test.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { RawSource } from 'webpack-sources';
+import MAGIC_NUMBER from 'metro/src/shared/output/RamBundle/magic-number';
+import FileRamBundle from '../FileRamBundle';
+
+test('FileRamBundle should create valid RAM bundle', () => {
+  const bootstrapper = '(function() { /* bootstrap */ })()';
+  const modules = [
+    {
+      id: 0,
```
Comment (thymikee) [inline packages/haul-ram-bundle-webpack-plugin/src/__tests__/FileRamBundle.test.ts]: likely not relevant to the test, but looks like `id` can be a string, maybe worth adding a case where it actually is a string

Comment (zamotany) [reply packages/haul-ram-bundle-webpack-plugin/src/__tests__/FileRamBundle.test.ts]: it's tested in integration tests

File: packages/haul-ram-bundle-webpack-plugin/src/__tests__/integration.test.ts [hunk 3]
```diff
@@ -213,30 +247,154 @@ describe('should build and successfully evaluate bundle', () => {
...
+
+    await wait(1);
+
+    expect(global.crash).toBeDefined();
+    expect(() => {
+      ((global as any) as { crash: Function }).crash();
+    }).toThrow('test error');
+  });
+
+  it('with namedModules:true', async () => {
```
Comment (thymikee) [inline packages/haul-ram-bundle-webpack-plugin/src/__tests__/integration.test.ts]: there's quite some duplicated code here. consider using `test.each` for less repetition

**Pull request merged**